src/librbd/api/Trash.cc

```diff
@@ -145,30 +138,64 @@ int Trash<I>::move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
     return -EOPNOTSUPP;
   }
 
-  image_id = ictx->id;
-  ictx->owner_lock.get_read();
-  if (ictx->exclusive_lock != nullptr) {
-    ictx->exclusive_lock->block_requests(0);
+  std::string id = ictx->id;
+  if(id.empty() && r == -ENOENT) {
```

Comment:
nit: if `id` is empty we can't proceed anyway so no need to check for -ENOENT (which should be alway true in this case anyway), just return `-ENOENT`.